addons/survey/models/res_lang.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, models, _
+from odoo.exceptions import UserError
+
+
+class ResLang(models.Model):
+    _inherit = "res.lang"
+
+    def write(self, vals):
+        """ When a language is disabled, clear corresponding survey language. """
+        if 'active' in vals and not vals['active']:
+            self.env['survey.user_input'].search([('lang_id', 'in', self.ids)]).lang_id = False
+            surveys_sudo = self.env['survey.survey'].search([('lang_ids', 'in', self.ids)]).sudo()
+            surveys_sudo_all_langs = surveys_sudo.filtered(lambda survey: not survey.lang_ids)
+            surveys_sudo.write({'lang_ids': [Command.unlink(lang.id) for lang in self]})
+            survey_sudo_all_langs_after_disabling_lang = surveys_sudo.filtered(lambda survey: not survey.lang_ids)
+            if impacted_surveys_sudo := survey_sudo_all_langs_after_disabling_lang - surveys_sudo_all_langs:
```

Comment:
Isn't it simply filtering `surveys_sudo.filtered(lambda s: len(s.lang_ids) == len(self))`, so that we could even check before `write`?